src/NUnitTestAdapterTests/Filtering/TestFilterTests.cs

```diff
@@ -78,7 +82,17 @@ public void PropertyValueProviderCategoryWithOneCategory()
         }
 
         [Test]
-        public void PropertyValueProviderCategoryWithNoTraits()
+        public void PropertyValueProviderWithOneCategoryAndCategoryAsFilter()
+        {
+            var tc = new TestCase("Test1", new Uri("executor://NUnitTestExecutor"), "NUnit.VSIX");
+            tc.AddTrait("Category", "CI");
```

Comment:
Unrelated: I wish we weren't testing by modifying the TestCase object directly because the *way* we apply categories to TestCases may change in the future. @OsirisTerje I'll try to rememeber this when I dust off the refactoring PR.